File: Problems/2706. Buy Two Chocolates/2706. Buy Two Chocolates.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        int[] smallest = {Integer.MAX_VALUE, Integer.MAX_VALUE};
        for (int price : prices) {
            smallest[1] = Math.min(price, smallest[1]);
            if (smallest[1] < smallest[0]) {
                int tmp = smallest[0];
                smallest[0] = smallest[1];
                smallest[1] = tmp;
            }
        }

        if (money-smallest[0]-smallest[1] < 0) return money;
        return money-smallest[0]-smallest[1];
    }
}

File: Problems/2706. Buy Two Chocolates/2706. Buy Two Chocolates.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        min_val = [float('inf'), float('inf')]
        
        for price in prices:
            if price < min_val[0]:
                min_val[1] = min_val[0]
                min_val[0] = price
            elif price < min_val[1]:
                min_val[1] = price
                
        return money-sum(min_val) if money-sum(min_val) >= 0 else money